### ISSUE TITLE
Setup a TG CI pipeline and run post commit tests

### ISSUE THREAD

**Issue by ubcheema** [tenstorrent/tt-metal#8452]
We need a TG CI machine that runs post commits tests specific to TG system.
Galaxy bringup is happening on branch: galaxy/main.
We should setup post commit test runs when changes are pushed to galaxy/main.
To start, we can enable single chip tests that run on mmio chip as well as remote chip.

**Comment by tapspatel** [comment]
One thing to note, it might be a good idea to continuously merge branch into main and not let it drift, so we stay updated to tip.

**Comment by tt-rkim** [comment]
This is needed, however please do not do what fd-2 did last time and have a giant, diverging branch. Please merge into main often as @tapspatel said.

We want to discourage this method of development, and encourage iterative development. You can gate things behind DEFINEs during compilation

@tapspatel will be looking into tg pipeline (should be main already with no tests)

**Comment by tapspatel** [comment]
Merged: dc8c35ace42e062a13d95b2d0701578f89c7d6b1

**Comment by aliuTT** [comment]
Currently, TG CI has 8 hugepages, need to upgrade to 16 (4 per MMIO). @tapspatel

**Comment by tapspatel** [comment]
fixing...

**Comment by tapspatel** [comment]
Fixed: 16 pages per TG and TGG (as UMD only supports 4 per mmio device currently). Will need
tg needs 36
tgg needs 72

once UMD support for it lands. closing.
